SamplifyAPIClient/Response.cs

```diff
@@ -232,6 +232,15 @@ public class GetSurveyTopicsResponse : Response
         public SurveyTopic[] Data { get; set; }
     }
 
+    [DataContract]
+    public class GetInvoicesSummaryResponse : Response
```

Comment:
would this cover error response? Below is the structure.
{
    "data": null,
    "meta": null,
    "status": {
        "errors": [
            {
                "code": "400",
                "message": "No projects found for the given input criteria",
                "resource": null
            },
            {
                "code": "INFO",
                "message": "requestId: pe-nt/rdGlK",
                "resource": null
            }
        ],
        "message": "bad request"
    }
}

Reply:
Good point.

Reply:
No. Initially I wanted to copy the Golang SDK's strategy. It does not account for a json response for pdf content. In my opinion, I think all the SDK behavior should act very similar.

I can change it is to handle a failed json request.

```
            if (!res.IsSuccessStatusCode)
            {
                string errPath = string.Format("{0}{1}", host, url);
                ErrorResponse err = new ErrorResponse();
                err.Timestamp = DateTime.Now;
                err.RequestID = reqID;
                err.HTTPCode = (int)res.StatusCode;
                err.HTTPPhrase = res.ReasonPhrase;
                err.Path = errPath;
                return new APIResponse(reqID, json, err);
            }

            if (res.Content.Headers.ContentType.MediaType == "application/pdf")
            {
                byte[] rawMsg = await res.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
                return new APIResponseRaw(reqID, rawMsg, null);
            }


            return new APIResponse(reqID, json, null);
```

Reply:
The response now will be a json or parse to models (covering error response). Else it will return bytes. I tried googling dotnet but I don't how to automatically parse an `application/pdf` Content-Type.